lib/node_modules/@stdlib/array/base/map/docs/types/index.d.ts

```diff
@@ -0,0 +1,214 @@
+
+/**
+* Interface describing the main export.
+*/
+interface Routine {
+	/**
+	* Applies a callback function to elements in an input array and assigns results to elements in a new output array.
+	*
+	* @param x - input array
+	* @param fcn - callback function
+	* @param thisArg - callback execution context
+	* @returns output array
+	*
+	* @example
+	* var ones = require( '@stdlib/array/base/ones' );
+	*
+	* function scale( v ) {
+	*     return v * 10;
+	* }
+	*
+	* var x = ones( 4 );
+	* var y = map( x, scale );
+	* // returns [ 10, 10, 10, 10 ]
+	*/
+	<T = unknown, U = unknown, V = unknown>( x: Collection<T>, fcn: Callback<T, U, V>, thisArg?: ThisParameterType<Callback<T, U, V>> ): Collection<U>;
+
+	/**
+	* Applies a callback function to elements in an input array and assigns results to elements in a new output array.
+	*
+	* @param x - input array object
+	* @param fcn - callback function
+	* @param thisArg - callback execution context
+	* @returns output array object
+	*
+	* @example
+	* var toAccessorArray = require( '@stdlib/array/to-accessor-array' );
+	* var ones = require( '@stdlib/array/base/ones' );
+	*
+	* function scale( v ) {
+	*     return v * 10;
+	* }
+	*
+	* var x = toAccessorArray( ones( 4 ) );
+	* var y = map( x, scale );
+	* // returns {...}
+	*/
+	<T = unknown, U = unknown, V = unknown>( x: AccessorArrayLike<T>, fcn: Callback<T, U, V>, thisArg?: ThisParameterType<Callback<T, U, V>> ): AccessorArrayLike<U>;
+
+	/**
+	* Applies a callback function to elements in an input array and assigns results to elements in an output array.
+	*
+	* @param x - input array
+	* @param y - output array
+	* @param fcn - callback function
+	* @param thisArg - callback execution context
+	* @returns output array
+	*
+	* @example
+	* var ones = require( '@stdlib/array/base/ones' );
+	* var zeros = require( '@stdlib/array/base/zeros' );
+	*
+	* function scale( v ) {
+	*     return v * 10;
+	* }
+	* var x = ones( 4 );
+	* var y = zeros( x.length );
+	*
+	* var out = map.assign( x, y, scale );
+	* // returns [ 10, 10, 10, 10 ]
+	*
+	* var bool = ( out === y );
+	* // returns true
+	*/
+	assign<T = unknown, U = unknown, V = unknown>( x: Collection<T>, y: Collection<U>, fcn: Callback<T, U, V>, thisArg?: ThisParameterType<Callback<T, U, V>> ): Collection<U>;
+
+	/**
+	* Applies a callback function to elements in an input array and assigns results to elements in an output array.
+	*
+	* @param x - input array
+	* @param y - output array
+	* @param fcn - callback function
+	* @param thisArg - callback execution context
+	* @returns output array
+	*
+	* @example
+	* var toAccessorArray = require( '@stdlib/array/to-accessor-array' );
+	* var ones = require( '@stdlib/array/base/ones' );
+	* var zeros = require( '@stdlib/array/base/zeros' );
+	*
+	* function scale( v ) {
+	*     return v * 10;
+	* }
+	* var x = toAccessorArray( ones( 4 ) );
+	* var y = toAccessorArray( zeros( x.length ) );
+	*
+	* var out = map.assign( x, y, scale );
+	* // returns {...}
```

Comment:
```suggestion
	* var out = map.assign( toAccessorArray( x ), toAccessorArray( y ), scale );
	* // y => [ 10, 10, 10, 10 ]
```

Show the expected results.